### PR TITLE
Allow property assignment and array operations from subqueries.

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/ast.py [hunk 1]
```diff
@@ -2950,14 +2964,32 @@ async def run(self, runt, genr):
...
+
+            valu = await self.kids[2].compute(runt, path)
+            valu = await s_stormtypes.toprim(valu)
+
+            # Setting a value to a subquery means assigning the value of
+            # the primary property (for a single value) or array of values
+            if isinstance(self.kids[2], SubQuery):
+
+                if not isarray:
+                    if len(valu) != 1:
```
Comment (Cisphyx) [inline synapse/lib/ast.py:2984]: Would we want to allow `?=` to work if the subquery returns no nodes?

File: synapse/lib/ast.py [hunk 2]
```diff
@@ -2950,14 +2964,32 @@ async def run(self, runt, genr):
...
+                        mesg = 'Cannot assign a property value from multiple nodes.'
```
Comment (Cisphyx) [inline synapse/lib/ast.py:2985]: This error message could be confusing if they get it due to the subquery returning no nodes.

File: synapse/lib/ast.py [hunk 3]
```diff
@@ -2950,14 +2964,32 @@ async def run(self, runt, genr):
         async for node, path in genr:
 
             name = await self.kids[0].compute(runt, path)
-
-            valu = await self.kids[2].compute(runt, path)
-            valu = await s_stormtypes.toprim(valu)
+            #valu = await self.kids[2].compute(runt, path)
```
Comment (Cisphyx) [inline synapse/lib/ast.py:2967]: commented out code

File: synapse/lib/ast.py (hunk 1 above)
Comment (vEpiphyte) [reply synapse/lib/ast.py:2984]: What would be the expected behavior for `?=` if there were >1 nodes?

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**